etcdserver/etcdhttp/httptypes/errors.go

```diff
@@ -43,7 +43,10 @@ func (e HTTPError) WriteTo(w http.ResponseWriter) {
 	if err != nil {
 		plog.Panicf("marshal HTTPError should never fail (%v)", err)
 	}
-	w.Write(b)
+	if _, err := w.Write(b); err != nil {
+		plog.Errorf("http write errors with HTTPError (%v)", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
I think the write error can only be caused by a broken connection or write timeout. Not sure if it is still reasonable to retry writing to the same client again. I would suggest just return here.